### ISSUE TITLE
listen on a port 

### ISSUE THREAD

**Issue by celian-hamon** [bwmarrin/discordgo#1189]
is the bot listening on a port ?

**Comment by celian-hamon** [comment]
sorry i need to deploy on google cloud but i cant without listen on a port

**Comment by FedorLap2006** [comment]
TL;DR No, bot (in particular gateway) doesn't listen on any port, instead it connects to a Discord's API server.

Also, my guess is that you're using cloud run/functions. They're meant only for servers (HTTP, gRPC, websockets) etc.
By just `Open`ing gateway session you aren't listening on any port. So in this case I would recommend to use k8s/compute cloud for that, as they provide environment that is running 24/7.

However, if your bot works on interactions (or just has them integrated, through gateway), you can migrate the bot/interactions code to cloud run/functions. To do that you need to setup an HTTP server on provided port and listen for interactions there. The URL (IP of the server + relative path you set in the server) you can specify on the Developer Portal.

For more info you can visit the official [documentation](https://discord.com/developers/docs/interactions/receiving-and-responding#receiving-an-interaction).
